src/Factory.php

```diff
@@ -573,4 +616,32 @@ private function isPersisting(): bool
 
         return true;
     }
+
+    /**
+     * @return TObject
+     */
+    private function instantiateAndHydrate(array $attributes): object
+    {
+        $instantiator = $this->instantiator ?? self::configuration()->instantiator();
+
+        if (!$instantiator instanceof Instantiator && !$this->hydrator) {
```

Comment:
> Note: if using a closure or non-Instantiator callable for the instantiator, the hydrator isn't used.

shouldn't we throw an error if we detect an hydrator which is not the default hydrator in this case?

Reply:
Perhaps but then we'd break the possibility of fixing cascade persist with custom instantiators. One problem at a time though. I'll throw an exception for now.

Reply:
damn, that's quite a problem :exploding_head: